api/backend.go

```diff
@@ -615,104 +614,49 @@ func appendIf(condition bool, services []gethnode.ServiceConstructor, service ge
 	return append(services, service)
 }
 
-// CreateContactCode create or return the latest contact code
-func (b *StatusBackend) CreateContactCode() (string, error) {
-	selectedChatAccount, err := b.AccountManager().SelectedChatAccount()
-	if err != nil {
-		return "", err
-	}
-
-	st, err := b.statusNode.ShhExtService()
-	if err != nil {
-		return "", err
-	}
-
-	bundle, err := st.GetBundle(selectedChatAccount.AccountKey.PrivateKey)
-	if err != nil {
-		return "", err
-	}
-
-	return bundle.ToBase64()
+// ExtractGroupMembershipSignatures extract signatures from tuples of content/signature
+func (b *StatusBackend) ExtractGroupMembershipSignatures(signaturePairs [][2]string) ([]string, error) {
+	return crypto.ExtractSignatures(signaturePairs)
 }
 
-// GetContactCode return the latest contact code
-func (b *StatusBackend) GetContactCode(identity string) (string, error) {
-	st, err := b.statusNode.ShhExtService()
-	if err != nil {
-		return "", err
-	}
-
-	publicKeyBytes, err := hex.DecodeString(identity)
-	if err != nil {
-		return "", err
-	}
-
-	publicKey, err := ethcrypto.UnmarshalPubkey(publicKeyBytes)
-	if err != nil {
-		return "", err
-	}
-
-	bundle, err := st.GetPublicBundle(publicKey)
+// SignGroupMembership signs a piece of data containing membership information
+func (b *StatusBackend) SignGroupMembership(content string) (string, error) {
+	selectedChatAccount, err := b.AccountManager().SelectedChatAccount()
 	if err != nil {
 		return "", err
 	}
 
-	if bundle == nil {
-		return "", nil
-	}
-
-	return bundle.ToBase64()
+	return crypto.Sign(content, selectedChatAccount.AccountKey.PrivateKey)
 }
 
-// ProcessContactCode process and adds the someone else's bundle
-func (b *StatusBackend) ProcessContactCode(contactCode string) error {
-	selectedChatAccount, err := b.AccountManager().SelectedChatAccount()
-	if err != nil {
-		return err
-	}
-
+// LoadFilters loads filter on sshext
+func (b *StatusBackend) LoadFilters(chats []*filter.Chat) ([]*filter.Chat, error) {
```

Comment:
hmm, can it go through json-rpc instead? at least, in my attemts, I envision all the protocol stuff called through json-rpc, so bringing these together will make it more consistent.

Reply:
also, no need to implement it in 2 places (mobile + library)